flight/Modules/VtolPathFollower/vtol_follower_fsm.c

```diff
@@ -546,46 +507,22 @@ static int32_t do_land()
  */
 static int32_t do_loiter()
 {
-	const float LOITER_LEASH = 4;
-
-	LoiterCommandData loiterCommand;
-	LoiterCommandGet(&loiterCommand);
-
-	float yaw;
-	AttitudeActualYawGet(&yaw);
-	yaw *= DEG2RAD;
+	float att_adj[2] = { 0, 0 };
+	if (vtol_follower_control_loiter(DT, vtol_hold_position_ned, att_adj)) {
+		// If we did something, enable this as our hold position.
 
-	float north_offset = 0;
-	float east_offset = 0;
-	float down_offset = 0;
-
-	if (loiterCommand.Frame == LOITERCOMMAND_FRAME_BODY) {
-		north_offset = (loiterCommand.Forward * cosf(yaw) - loiterCommand.Right * sinf(yaw)) * DT;
-		east_offset = (loiterCommand.Forward * sinf(yaw) + loiterCommand.Right * cosf(yaw)) * DT;
-	} else {
-		north_offset = loiterCommand.Forward * DT;
-		east_offset = loiterCommand.Right * DT;
+		hold_position(vtol_hold_position_ned[0],
+				vtol_hold_position_ned[1],
+				vtol_hold_position_ned[2]);
 	}
```

Comment:
Actually it's not impossible it may want to tweak attitude even when not updating the hold position.